File: tests/benchmarks/resolution/fixtures/swift/Repository.swift
import Foundation

protocol Repository {
    associatedtype Entity
    func findById(_ id: String) -> Entity?
    func save(_ entity: Entity)
    func delete(_ id: String) -> Bool
}

class UserRepository: Repository {
    typealias Entity = User

    private var store: [String: User] = [:]

    func findById(_ id: String) -> User? {
        return store[id]
    }

    func save(_ user: User) {
        store[user.id] = user
    }

    func delete(_ id: String) -> Bool {
        return store.removeValue(forKey: id) != nil
    }

    func count() -> Int {
        return store.count
    }
}

struct User {
    let id: String
    let name: String
    let email: String
}

File: tests/benchmarks/resolution/fixtures/scala/Repository.scala
case class User(id: String, name: String, email: String)

class UserRepository {
  def findById(id: String): Option[User] = {
    None
  }

  def save(user: User): Unit = {
    println(user.id)
  }

  def delete(id: String): Boolean = {
    false
  }
}
